core/trino-main/src/main/java/io/trino/operator/scalar/DateTimeFunctions.java

```diff
@@ -308,34 +309,69 @@ public static long diffDate(@SqlType("varchar(x)") Slice unit, @SqlType(Standard
         return getDateField(UTC_CHRONOLOGY, unit).getDifferenceAsLong(DAYS.toMillis(date2), DAYS.toMillis(date1));
     }
 
+    /**
+     * Data tuple to store mapping between date time unit and date fraction extraction logic.
+     * This record is intended to be used for a specific use case only.
+     *
+     * @param unit date time unit represented as ASCII bytes
+     * @param dateTimeFieldProvider callback to extract fraction of date
+     */
+    private record DateTimeFieldProvider(byte[] unit, Function<ISOChronology, DateTimeField> dateTimeFieldProvider)
+    {
+    }
+
+    private static final DateTimeFieldProvider[] DATE_FIELDS = new DateTimeFieldProvider[] {
+            new DateTimeFieldProvider("day".getBytes(US_ASCII), ISOChronology::dayOfMonth),
+            new DateTimeFieldProvider("week".getBytes(US_ASCII), ISOChronology::weekOfWeekyear),
+            new DateTimeFieldProvider("month".getBytes(US_ASCII), ISOChronology::monthOfYear),
+            new DateTimeFieldProvider("quarter".getBytes(US_ASCII), QUARTER_OF_YEAR::getField),
+            new DateTimeFieldProvider("year".getBytes(US_ASCII), ISOChronology::year)
+    };
+    private static final DateTimeFieldProvider[] TIMESTAMP_FIELDS = new DateTimeFieldProvider[] {
+            new DateTimeFieldProvider("millisecond".getBytes(US_ASCII), ISOChronology::millisOfSecond),
+            new DateTimeFieldProvider("second".getBytes(US_ASCII), ISOChronology::secondOfMinute),
+            new DateTimeFieldProvider("minute".getBytes(US_ASCII), ISOChronology::minuteOfHour),
+            new DateTimeFieldProvider("hour".getBytes(US_ASCII), ISOChronology::hourOfDay),
+            new DateTimeFieldProvider("day".getBytes(US_ASCII), ISOChronology::dayOfMonth),
+            new DateTimeFieldProvider("week".getBytes(US_ASCII), ISOChronology::weekOfWeekyear),
+            new DateTimeFieldProvider("month".getBytes(US_ASCII), ISOChronology::monthOfYear),
+            new DateTimeFieldProvider("quarter".getBytes(US_ASCII), QUARTER_OF_YEAR::getField),
+            new DateTimeFieldProvider("year".getBytes(US_ASCII), ISOChronology::year)
+    };
+
+    private static boolean equalsIgnoreCaseAscii(Slice slice, byte[] bytes)
```

Comment:
SliceUtf8.toLower converts a Slice to lower case. Then you can compare them using .equals()